From aab2b6e02f8f556c55834f0bc528c4d7cb2949a4 Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@nabladev.com>
Date: Mon, 29 Dec 2025 19:04:30 +0100
Subject: [PATCH 951/954] fix(st): avoid enormous tf-a-stm32mp13*stm32

The current STM32MP13xx build produces tf-a-stm32mp13*stm32 which is
about 800 MiB is size. This is caused by default build-id emitted by
GNU linker and no-pie build. Change the linker flags such that the
build is static and no build-id is generated. This produces expected
binary, with the right size, which works on the actual hardware again.
This fix is similar to 72f4b70e8e8e ("fix(rcar-layout): fix tool build").

This failure is triggered using most of arm-linux-* toolchains, which
are generated by OE-core 5.0 Scarthgap, arm-linux-gnueabi* 15.2.0 in
Debian and such toolchain. The arm-none-eabi- toolchains do not seem
to trigger this.

Upstream-Status: Backport [b66160f43b9dbc9a1617d94d6aab61ea5bc37c12]
Change-Id: I0f7d9f330dbe08d6adf491ed79b3a10a4adfecc7
Fixes: 6d26d75c374b ("fix(st): set no-pie option when building ST elf file")
Signed-off-by: Marek Vasut <marex@nabladev.com>
---
 plat/st/common/common_rules.mk | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/plat/st/common/common_rules.mk b/plat/st/common/common_rules.mk
index 5cc3c61062..2931790524 100644
--- a/plat/st/common/common_rules.mk
+++ b/plat/st/common/common_rules.mk
@@ -59,9 +59,9 @@ $(eval $(call MAKE_LD,${STM32_TF_LINKERFILE},$(STM32_LD_FILE),bl2))
 tf-a-%.elf: $(PLAT)-%.o ${STM32_TF_LINKERFILE}
 	@echo "  LDS     $<"
 ifneq ($(findstring gcc,$(notdir $(LD))),)
-	${Q}${LD} -o $@ $(subst --,-Wl$(comma)--,${STM32_TF_ELF_LDFLAGS}) -nostartfiles -no-pie -Wl,-Map=$(@:.elf=.map) -Wl,-dT ${STM32_TF_LINKERFILE} $<
+	${Q}${LD} -o $@ $(subst --,-Wl$(comma)--,${STM32_TF_ELF_LDFLAGS}) -nostartfiles -static -Wl,--build-id=none -Wl,-Map=$(@:.elf=.map) -Wl,-dT ${STM32_TF_LINKERFILE} $<
 else
-	${Q}${LD} -o $@ ${STM32_TF_ELF_LDFLAGS} -no-pie -Map=$(@:.elf=.map) --script ${STM32_TF_LINKERFILE} $<
+	${Q}${LD} -o $@ ${STM32_TF_ELF_LDFLAGS} --static --build-id=none -Map=$(@:.elf=.map) --script ${STM32_TF_LINKERFILE} $<
 endif
 
 tf-a-%.bin: tf-a-%.elf
-- 
2.53.0

